templates/project/src/main.rs

```diff
@@ -55,6 +55,8 @@ fn main() {
     esp_idf_svc::sys::link_patches();
     initialize_logger::<EspLogger>();
 
+    log::info!("{} started (esp32)", env!("CARGO_PKG_NAME"));
```

Comment:
I believe this gets evaluated at build time, so this will nicely report the name of the project generated from the template.